utils/src/utils/Environment.cpp

```diff
@@ -123,4 +130,49 @@ bool Environment::isRunningAsService() {
   return runningAsService;
 }
 
+std::unordered_map<std::string, std::string> Environment::getEnvironmentVariables() {
+  std::unordered_map<std::string, std::string> env_var_map;
+
+#ifdef WIN32
+  LPWCH env_strings = GetEnvironmentStringsW();
+  if (!env_strings) {
+    return env_var_map;
+  }
+
+  LPWCH env = env_strings;
+
+  while (*env) {
+    std::wstring wstring_variable_key_value_pair(env);
+
+    int size_needed = WideCharToMultiByte(CP_UTF8, 0, wstring_variable_key_value_pair.c_str(), -1, nullptr, 0, nullptr, nullptr);
+    std::vector<char> buffer(size_needed);
+    WideCharToMultiByte(CP_UTF8, 0, wstring_variable_key_value_pair.c_str(), -1, buffer.data(), size_needed, nullptr, nullptr);
+
+    std::string variable_key_value_pair(buffer.data());
```

Comment:
This one should pass the size of the converted string too to the string constructor.